vscode_extension/src/LanguageClient.ts

```diff
@@ -317,7 +323,15 @@ export default class SorbetLanguageClient implements ErrorHandler {
   public closed(): CloseAction {
     if (this._status !== ServerStatus.ERROR) {
       this._updateStatus(ServerStatus.RESTARTING);
-      this._restart(RestartReason.CRASH);
+      let reason = RestartReason.CRASH_LC_CLOSED;
+      if (this._sorbetProcessExitCode === 11) {
+        // 11 is EAGAIN
+        reason = RestartReason.WRAPPER_REFUSED_SPAWN;
+      } else if (this._sorbetProcessExitCode === 143) {
+        // 143 = 128 + 15 and 15 is TERM signal
+        reason = RestartReason.FORCIBLY_TERMINATED;
+      }
```

Comment:
Are these the Linux values or the Mac values?  My Linux box agrees that `EAGAIN` is 11, but my Mac headers say that 11 is `EDEADLK` ("11 was EAGAIN" in a comment) and 35 is `EAGAIN`.

`SIGTERM` does appear to be the same on both.

(Citing for completeness, I'm not sure it's worth trying to be perfectly correct here.)

Reply:
interesting i didn't realize they were different. The only thing that actually matters here is the number that we choose in the wrapper script, which can be anything. I had chosen 11 because it was EAGAIN on linux and seemed like it would be unlikely to ever collide with any return value returned from sorbet itself.

Thanks for reminding me, I need to send a PR to Stripe's codebase to land that wrapper script.